.github/workflows/release.yaml

```diff
@@ -0,0 +1,79 @@
+---
+name: Release
+
+run-name: release ${{ inputs.tag }}
+
+on:
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: 'Tag to be released'
+        required: true
+
+permissions:
+  packages: write
+  id-token: write
+  contents: read
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Set env vars
+        run: |
+          echo PKG_VERSION="$(yarn -s run get-version)" >> $GITHUB_ENV
+          echo PKG_NAME="$(yarn -s run get-package-name)" >> $GITHUB_ENV
+          echo PKG_BASENAME="$(yarn -s run get-package-basename)" >> $GITHUB_ENV
+          echo VERSION="${{ inputs.tag }}" >> $GITHUB_ENV
+      - name: Ensure the branch is protected
+        run: |
+          if [[ "${{ github.ref_protected }}" == "false" ]]; then
+            echo "::error::The branch ${{ github.ref }} is not protected"
+            exit 1
+          fi
+      - name: Ensure tag has not been released
+        run: |
+          TAG_EXISTS=$(git tag -l "${{ inputs.tag }}")
+          if [[ -n "$TAG_EXISTS" ]]; then
+            echo "::error::The tag ${{ inputs.tag }} already exists"
+            exit 1
+          fi
+      - name: Ensure version is properly set
+        run: |
+           if ! [[ "$PKG_VERSION" = "$VERSION" ]]; then
+             echo "::error file=package.json,line=$(sed -n '/"version":/=' package.json)::The tag $VERSION must match the version $PKG_VERSION specified in package.json"
+             exit 1
+           fi
+      - name: Setup node with GitHub Packages
+        uses: actions/setup-node@v4
+        with:
+          cache: yarn
+          node-version: '16'
+          registry-url: https://npm.pkg.github.com
```

Comment:
should be specified in `package.json` instead

Reply:
We must do it this way for public repos. For private we can set them on GitHub, but we'll still need to specify `registry-url` on the action as it does some `.npmrc` configuration for us.